### ISSUE TITLE
Stopped working even if logged in 

### ISSUE THREAD

**Issue by amacgregor** [znck/grammarly#183]
Started getting the following error:

```
1618161793982 DEBUG  [TextGrammarCheckHost] (66696c) Hosting file:///Volumes/Workspace/Writing/Draft.dev/in-progress/Achieving%20Repeatability%20in%20Continuous%20Integration/Request.md
(node:392999) UnhandledPromiseRejectionWarning: Error: Socket connection failed: NOT_AUTHORIZED
    at GrammarlyClient._connect (/home/amacgregor/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38903:19)
    at GrammarlyClient._queueOrSend (/home/amacgregor/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38978:18)
    at /home/amacgregor/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38876:18
    at new Promise (<anonymous>)
    at GrammarlyClient.send (/home/amacgregor/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38874:16)
    at GrammarlyClient.start (/home/amacgregor/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39160:21)
    at GrammarlyClient.onConnection [as _onConnection] (/home/amacgregor/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39809:32)
(node:392999) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
```

**Comment by andylee830914** [comment]
I got the same error these days.

**Comment by Shisagi** [comment]
I am getting the same error.

**Comment by paolino** [comment]
UnhandledPromiseRejectionWarning: Error: Socket connection failed: NOT_AUTHORIZED

What to do?

Ironically, it's helping to write this comment :-|

**Comment by lucasfijen** [comment]
Getting the same error

**Comment by znck** [comment]
I'll try look into this today.

**Comment by znck** [comment]
Couldn't reproduce. Need more info.

**Comment by beyefendi** [comment]
```
(node:49088) UnhandledPromiseRejectionWarning: Error: Socket connection failed: NOT_AUTHORIZED
    at GrammarlyClient._connect (/home/emre/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38903:19)
    at GrammarlyClient._queueOrSend (/home/emre/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38978:18)
    at /home/emre/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38876:18
    at new Promise (<anonymous>)
    at GrammarlyClient.send (/home/emre/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:38874:16)
    at GrammarlyClient.start (/home/emre/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39160:21)
    at GrammarlyClient.onConnection [as _onConnection] (/home/emre/.vscode/extensions/znck.grammarly-0.14.0/dist/server/index.js:39809:32)
(node:49088) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
```

**Comment by arnthorm** [comment]
I was facing the same problem, I logged out (Grammarly: Logout of grammarly.com) and then it started to work again and was still working when I logged back in.

Maybe the token expired?

**Comment by ssbarnea** [comment]
Any updates on this, the "login" popups that we cannot do anything about are a serious distraction.

**Comment by michidk** [comment]
Getting the same error:
```
(node:16276) UnhandledPromiseRejectionWarning: Error: Socket connection failed: NOT_AUTHORIZED
    at GrammarlyClient._connect (c:\Users\michi\.vscode\extensions\znck.grammarly-0.14.0\dist\server\index.js:38903:19)
    at GrammarlyClient._queueOrSend (c:\Users\michi\.vscode\extensions\znck.grammarly-0.14.0\dist\server\index.js:38978:18)
    at c:\Users\michi\.vscode\extensions\znck.grammarly-0.14.0\dist\server\index.js:38876:18
    at new Promise (<anonymous>)
    at GrammarlyClient.send (c:\Users\michi\.vscode\extensions\znck.grammarly-0.14.0\dist\server\index.js:38874:16)
    at GrammarlyClient.start (c:\Users\michi\.vscode\extensions\znck.grammarly-0.14.0\dist\server\index.js:39160:21)
    at GrammarlyClient.onConnection [as _onConnection] (c:\Users\michi\.vscode\extensions\znck.grammarly-0.14.0\dist\server\index.js:39809:32)
(node:16276) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

Logging out and in again does not help. It works fine though on a different workstation.

**Comment by znck** [comment]
The `NOT_AUTHORIZED` error code occurs when access token received from OAuth has expired. I haven't implemented token deletion or refreshing. However, as a workaround you can:

1. **Logout** using `Grammarly: Logout from grammarly.com`
2. **Reload**  or **Restart**  VS Code
3. **Login** using `Grammarly: Login to grammarly.com`

I don't have bandwidth to fix this issue right now but I am open to accepting PRs.

**Comment by znck** [comment]
Fixed in v0.18.0
